### ISSUE TITLE
Undo is not working on RadioButton

### ISSUE THREAD

**Issue by pashminakazi** [apache/royale-asjs#678]
Project Details are :
https://paste.apache.org/cpo8y
Test case is available on VDI.
Go to Import from Select in Menu
Go to Import -> Vessel from Setups in Menu
File : com/dbz/modules/IMP/dbzVesselSetup.mxml

![Undo on RadioButton](https://user-images.githubusercontent.com/42200979/72333723-16b7b800-36de-11ea-9789-0cc1a4800c9e.gif)

**Comment by aharui** [comment]
Please explain in words what is the problem, what components are involved.  I cannot tell from watching the video.

**Comment by pashminakazi** [comment]
Thanks for your help Alex,
Issue is :
if ADG has no records then the vertical scrollbar should not be visible in
ADG
if ADG has some records which is shown on Screen and we do not need to
scroll down to see these records in ADG then the vertical scrollbar should
not be visible in ADG
if ADG has more records and we need scroller to see them then Vertical
Scrollbar is needed otherwise not.

**Comment by aharui** [comment]
This issue seems to be about RadioButton, not ADG

**Comment by pashminakazi** [comment]
oh sorry i was thinking that you were asking about Vertical Scrollbar,
Yes this issues is in RadioButton

**Comment by aharui** [comment]
In words, what is expected to happen and what is happening?

**Comment by pashminakazi** [comment]
When we select 1st Radiobutton and then select 2nd radiobutton and click
undo button the Selected radiobutton should be 1st

**Comment by pashminakazi** [comment]
Not working still i am unable to undo radiobutton

**Comment by pashminakazi** [comment]
In both Cases undo should work
In the Case of New -> It is working when we click new button and select radiobutton (1) and then select radiobutton(2) and then click undo button.Selected radiobutton is (1)

but in case of saved record it is not working.
if I have selected radiobutton (1) and save a record then i click radiobutton(2) and click undo it doesn't work

**Comment by aharui** [comment]
Did you debug the undo code?  Does it set something on the radio buttons or does it rely on setting the text of purchaseRequColor, and does it set it to the correct value?

**Comment by aharui** [comment]
I debugged your undo code as I suggested 8 days ago.  As currently written, the Undo code will not reset the radio buttons for a saved item.  The fn_Undo calls PopulateFormWithDGItem, which checks for `dg.selectedItem == null`.  It is null for a new item so it calls fn_ResetPage.  For a saved item, it calls setForm.  However, the arrFormItems passed to setForm does not contain the radioButtons because the radio buttons are children of a BorderContainer and arrFormItems is populated by top-level components.

I do not know how this could work in Flex.  Maybe the radio buttons were not in  a BorderContainer or were otherwise added to arrFormItems, but I also do not see handling for RadioButtons in setForm.  Are you sure this code is exactly the same as the Flex code?
